### ISSUE TITLE
False positive case for indent rule

### ISSUE THREAD

**Issue by apb7** [eslint/eslint#10106]
**Tell us about your environment**

* **ESLint Version:**  
4.19.0
* **Node Version:**  
6.9.1
* **npm Version:**  
8.9.4

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": [
    "eslint:recommended",
  ],
  "plugins": [
    "angular",
    "html"
  ],
  "rules": {
    "angular/di-unused": 2,
    "angular/no-inline-template": [
      "error",
      {
        "allowSimple": true
      }
    ],
    "brace-style":[
      "error",
      "1tbs"
    ],
    "camelcase": [
      "error",
      {
        "properties": "never"
      }
    ],
    "comma-spacing": [
      "error",
      {
        "before": false,
        "after": true
      }
    ],
    "curly": [
      "error",
      "all"
    ],
    "dot-notation": [
      "error",
      {
        "allowKeywords": false
      }
    ],
    "eqeqeq": [
      "error",
      "always"
    ],
    "func-style": [
      "error",
      "expression"
    ],
    "indent": [
      "error",
      2,
      {
        "ArrayExpression": "first",
        "FunctionExpression": {
          "body": 1,
          "parameters": 2
        },
        "MemberExpression": 1,
        "SwitchCase": 1
      }
    ],
    "keyword-spacing": [
      "error",
      {
        "before": true,
        "after": true
      }
    ],
    "max-len": [
      "error",
      "code": 80,
      "ignoreUrls": true
    ],
    "no-console": [
      "error",
      {
        "allow": [
          "error",
          "warn"
        ]
      }
    ],
    "no-compare-neg-zero": "off",
    "no-constant-condition": [
      "off"
    ],
    "no-empty": [
      "error",
      {
        "allowEmptyCatch": true
      }
    ],
    "no-multi-spaces": [
      "error"
    ],
    "no-multi-str": [
      "error"
    ],
    "no-redeclare": [
      "off"
    ],
    "no-trailing-spaces": [
      "error"
    ],
    "no-undef": [
      "off"
    ],
    "no-unused-vars": [
      "off"
    ],
    "no-useless-escape": "off",
    "one-var": [
      "off"
    ],
    "operator-linebreak": [
      "error",
      "after"
    ],
    "padded-blocks": [
      "error",
      "never"
    ],
    "quotes": [
      "error",
      "single",
      {
        "avoidEscape": true
      }
    ],
    "quote-props": [
      "error",
      "as-needed",
      {
        "keywords": true
      }
    ],
    "semi": [
      "error",
      "always"
    ],
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "ignore",
        "named": "never",
        "asyncArrow": "ignore"
      }
    ],
    "space-infix-ops": [
      "error"
    ],
    "space-unary-ops": [
      "error",
      {
        "words": true,
        "nonwords": false,
      }
    ],
    "wrap-iife": [
      "error",
      "any"
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

We were trying to switch from Eslint version 3.18.0 to Eslint 4.19.0 at our organization to incorporate the latest features and bug fixes since we were behind by a major release.

```js
$scope.graphOptions = [{
  text: 'Labeled',
  option: 'isLabeled'
},
{
  text: 'Directed',
  option: 'isDirected'
},
{
  text: 'Weighted',
  option: 'isWeighted'
}];
```

```bash
eslint ./test.js
```

**What did you expect to happen?**
No indentation errors.

**What actually happened? Please include the actual, raw output from ESLint.**
```
   5:1  error  Expected indentation of 23 spaces but found 0  indent
   6:1  error  Expected indentation of 25 spaces but found 2  indent
   7:1  error  Expected indentation of 25 spaces but found 2  indent
   8:1  error  Expected indentation of 23 spaces but found 0  indent
   9:1  error  Expected indentation of 23 spaces but found 0  indent
  10:1  error  Expected indentation of 25 spaces but found 2  indent
  11:1  error  Expected indentation of 25 spaces but found 2  indent
  12:1  error  Expected indentation of 23 spaces but found 0  indent

✖ 8 problems (8 errors, 0 warnings)
  8 errors, 0 warnings potentially fixable with the `--fix` option.

```

**Comment by platinumazure** [comment]
Hi @apb7, thanks for the issue.

I'm not sure I understand. It seems to me that with your configuration of `"ArrayExpression": "first"`, the rule is working as designed:

```js
// Start of 1st elem --v
$scope.graphOptions = [{
  text: 'Labeled',
  option: 'isLabeled'
},
{
  text: 'Directed',
  option: 'isDirected'
},
{
  text: 'Weighted',
  option: 'isWeighted'
}];
```

Something like this should satisfy the rule as currently configured:

```js
$scope.graphOptions = [
{
  text: 'Labeled',
  option: 'isLabeled'
},
{
  text: 'Directed',
  option: 'isDirected'
},
{
  text: 'Weighted',
  option: 'isWeighted'
}];
```

**Comment by apb7** [comment]
Hi @platinumazure! Got it. Thank you :)
